python-mini-projects/projects/Image_watermark/watermark.py

```diff
@@ -0,0 +1,53 @@
+import os
+from PIL import Image
+from PIL import ImageFilter
+
+
+def watermark_photo(input_image_path, output_image_path, watermark_image_path):
+
+    base_image = Image.open(input_image_path)
+    watermark = Image.open(watermark_image_path)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *watermark*. The resource is allocated by call *Image.open*. Currently, there are execution paths that do not contain closure statements, for example, when *resize()* throws an exception. To prevent this resource leak, close *watermark* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).